src/vs/workbench/contrib/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -1041,4 +1044,93 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 	dispose() {
 		this._disposables = dispose(this._disposables);
 	}
+
+	// {{SQL CARBON EDIT}}
+	private promptADSLaunchRecommendedExtensions() {
+		const storageKey = 'extensionsAssistant/AppLaunchRecommendationsIgnore';
+
+		if (this.storageService.getBoolean(storageKey, StorageScope.GLOBAL, false)) {
+			return;
+		}
+
+		let recommendations: IExtensionRecommendation[];
+		let localExtensions: ILocalExtension[];
+		const getRecommendationPromise = this.getAppLaunchRecommendations().then(recs => { recommendations = recs; });
+		const getLocalExtensionPromise = this.extensionsService.getInstalled(ExtensionType.User).then(local => { localExtensions = local; });
+		Promise.all([getRecommendationPromise, getLocalExtensionPromise]).then(() => {
+			if (!recommendations.every(rec => { return localExtensions.findIndex(local => local.identifier.id.toLocaleLowerCase() === rec.extensionId.toLocaleLowerCase()) !== -1; })) {
+				return new Promise<void>(c => {
+					this.notificationService.prompt(
```

Comment:
I'm worried about showing another notification on 1st launch.  I think we need to rethink the experience to consolidate all the prompts we want to show in this scenario.